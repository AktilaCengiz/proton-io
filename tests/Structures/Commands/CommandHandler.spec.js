/* eslint-disable no-undef */
require("dotenv/config");
const CommandHandler = require("../../../src/Structures/Commands/CommandHandler");
const ProtonClient = require("../../../src/Structures/ProtonClient");
const checkEqualKeys = require("../../checkEqualKeys");

const directory = `${__dirname}/CommandModules`;

const client = new ProtonClient({
    intents: []
});

test("properties", () => {
    const commandHandler = new CommandHandler(client, {
        automateCategories: true,
        defaultCooldown: 10000,
        ignoreBots: true,
        ignoreSelf: true,
        prefix: "?",
        directory
    });
    commandHandler.loadAll();
    checkEqualKeys(commandHandler, {
        prefix: "?",
        defaultCooldown: 10000,
        ignoreBots: true,
    });
    expect(commandHandler).toMatchSnapshot();
});
