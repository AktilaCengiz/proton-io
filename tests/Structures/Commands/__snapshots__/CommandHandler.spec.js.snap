// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properties 1`] = `
CommandHandler {
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "aliasManager": AliasManager {
    "cache": Array [],
  },
  "automateCategories": true,
  "client": Object {
    "application": null,
    "channels": Array [],
    "guilds": Array [],
    "owners": Array [],
    "presence": undefined,
    "shard": null,
    "user": null,
    "users": Array [],
  },
  "cooldownManager": CooldownManager {
    "cache": Array [],
  },
  "defaultClientPermissions": null,
  "defaultCooldown": 10000,
  "defaultRateLimit": null,
  "defaultTyping": false,
  "defaultUserPermissions": null,
  "directory": "/home/heiwa/wioenena/workspace/packages/javascript/node/proton-io/tests/Structures/Commands/CommandModules",
  "ignoreBots": true,
  "ignoreSelf": true,
  "modules": Array [
    Object {
      "aliases": null,
      "category": "Users",
      "client": Object {
        "application": null,
        "channels": Array [],
        "guilds": Array [],
        "owners": Array [],
        "presence": undefined,
        "shard": null,
        "user": null,
        "users": Array [],
      },
      "clientPermissions": null,
      "cooldown": null,
      "executable": true,
      "filepath": "/home/heiwa/wioenena/workspace/packages/javascript/node/proton-io/tests/Structures/Commands/CommandModules/PingCommand.js",
      "handler": [Circular],
      "id": "Ping",
      "information": null,
      "ownerOnly": false,
      "rateLimit": null,
      "typing": null,
      "userPermissions": null,
    },
  ],
  "prefix": "?",
  Symbol(kCapture): false,
}
`;
